src/main/java/org/folio/consortia/repository/SharingRoleRepository.java

```diff
@@ -11,19 +11,28 @@
 
 public interface SharingRoleRepository extends JpaRepository<SharingRoleEntity, UUID> {
 
-  List<SharingRoleEntity> findByRoleId(UUID roleId);
+  List<SharingRoleEntity> findByRoleName(String roleName);
 
-  SharingRoleEntity findByRoleIdAndTenantId(UUID roleId, String tenantId);
+  SharingRoleEntity findByRoleNameAndTenantId(String roleName, String tenantId);
 
-  @Query("SELECT sr.tenantId FROM SharingRoleEntity sr WHERE sr.roleId = ?1")
-  Set<String> findTenantsByRoleId(UUID roleId);
+  @Query("SELECT sr.tenantId FROM SharingRoleEntity sr WHERE sr.roleName = ?1")
+  Set<String> findTenantsByRoleName(String roleName);
 
-  Set<String> findTenantsByRoleIdAndIsCapabilitySetsSharedTrue(UUID roleId);
+  @Query("SELECT sr.tenantId FROM SharingRoleEntity sr WHERE sr.roleName = ?1 and sr.isCapabilitiesShared = true")
+  Set<String> findTenantsByRoleNameAndIsCapabilitiesSharedTrue(String roleName);
 
-  Set<String> findTenantsByRoleIdAndIsCapabilitiesSharedTrue(UUID roleId);
+  @Query("SELECT sr.tenantId FROM SharingRoleEntity sr WHERE sr.roleName = ?1 and sr.isCapabilitySetsShared = true")
+  Set<String> findTenantsByRoleNameAndIsCapabilitySetsSharedTrue(String roleName);
+
+  @Query("select s.roleId from SharingRoleEntity s where s.roleName = ?1 and s.tenantId = ?2")
+  UUID findRoleIdByRoleNameAndTenantId(String roleName, String tenantId);
```

Comment:
this should work even without @ query annotation

Reply:
Taking time to verify whether it works or not, and also had some issue with it. using query is more convient and extra exlpanation for action of method. If there is strong reason to remove, for me keeping query is more good option

Reply:
it is ok to remain as is